### PR TITLE
Integration tests & N+1

### PR DESCRIPTION
**Pull request by myaserkhan**
**In this PR, I did the following:**

Solved N+1 problem when fetching all posts and their comments for a user without using the bullet gem

Used Capybara to write integration tests for each view of the Blog App project according to the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md)

  - Implemented tests for user index page
  - Implemented tests for user show page
  - Implemented tests for post index page
  - Implemented tests for post show page

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -28,7 +28,8 @@ def create
   private
 
   def find_user
-    @user = User.find(params[:user_id])
+    # @user = User.find(params[:user_id])
```
Comment (julie-ify) [inline app/controllers/posts_controller.rb]: - [x] Kindly remove this commented code.

File: spec/integration/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    it 'redirects to the post show page on clicking a user post' do
+      # post_link = user_post_path(user_id: @user.id, id: @post.id)
+      # find("a[href='#{post_link}']").click
+      click_link(@post.text)
+      expect(page).to have_current_path(user_post_path(@user, @post))
+    end
+
+    it 'redirects to open all posts of a user' do
+      # all_posts_link = user_posts_path(user_id: @user.id)
+      # find("a[href='#{all_posts_link}']").click
```
Comment (julie-ify) [inline spec/integration/user_show_spec.rb]: - [ ] Kindly remove commented code.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @myaserkhan and @OpondoG,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @myaserkhan 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  
 - All changes requested have been implemented 🚀  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**